websites/serializers.py

```diff
@@ -157,7 +157,68 @@ class Meta:
         fields = read_only_fields
 
 
-class WebsiteContentDetailSerializer(serializers.ModelSerializer):
+class WebsiteContentDetailSerializerMixin:
+    @property
+    def website(self):
+        """Fetch Website from name in URL"""
+        website_name = self.context["view"].kwargs["parent_lookup_website"]
+        return Website.objects.get(name=website_name)
+
+    def _validate_metadata_to_config(self, attrs, content_type):
+        """
+        Validate metadata payload for creating or editing WebsiteContent according to the starter config
+
+        Args:
+            attrs (dict): Payload from REST API
+            content_type (str): Type of content to validate
+        """
+        website = self.website
+        if not website.starter:
+            raise ValidationError(
+                "Website is missing a starter which is required for validation"
+            )
+        collections = website.starter.config["collections"]
+
+        try:
+            item = [_item for _item in collections if _item["name"] == content_type][0]
+        except IndexError:
+            raise ValidationError(
+                f"Unable to find field config for content type {content_type}"
+            )
+
+        fields = {field["name"]: field for field in item["fields"]}
+        metadata = attrs.get("metadata", {})
+        for name, value in metadata.items():
+            if name not in fields:
+                raise ValidationError(
+                    f"{name} not listed as a field in starter config for {content_type}"
+                )
+
+        for field_name, field in fields.items():
+            if field_name not in metadata:
+                continue
+            if field_name in ("markdown", "file", "title"):
+                raise ValidationError(f"Field {field_name} cannot go in metadata")
+
+            value = metadata[field_name]
+            widget = field["widget"]
+            if widget in ("text", "string", "markdown"):
+                if not isinstance(value, str):
+                    raise ValidationError(f"Expected string for {field_name}")
+            elif widget == "select":
+                if field.get("multiple"):
+                    if not isinstance(value, list):
+                        raise ValidationError(f"Expected list for {field_name}")
+                else:
+                    if not isinstance(value, (str, type(None))):
+                        raise ValidationError(
+                            f"Expected string or null for {field_name}"
+                        )
```

Comment:
Most/all of these validations seem to be for request data that should not be possible coming from the front end. I can think of a scenario where the data we get does not match the config fields – a user loads a content form with some site config, that config is updated on the backend, then the user submits the form. That's a larger issue that we need to address comprehensively, though. Adding this validation is jumping the gun, IMO. Lmk if I'm missing something